### DISCUSSION TITLE
how to disable sign up!

### DISCUSSION THREAD

**Discussion by mollynmo** [usememos/memos#3628] [Q&A] [answered]
Hi，New version, how to close registration.

**Comment by eMUQI** [comment]
![image](https://github.com/usememos/memos/assets/32332228/17c863fe-bf22-4c4e-b8cd-dd56114f4f15)

**Reply by slyang40** [reply]
It seems that this options is not available. See below attached screenshot. do I miss something in config? Running memos on synology nas.

![image](https://github.com/usememos/memos/assets/127069096/298a7987-f773-4799-b72f-bf819f90362e)

**Reply by thouseef** [reply, marked as answer]
seems like if you set an ENV variable `MEMOS_PUBLIC=false` and restart the container, signup gets disabled. 
Refer [Startup Options](https://www.usememos.com/docs/install/startup-options)

**Reply by slyang40** [reply]
@thouseef thanks for your information. It works after I setup the env variable, a behavior change in version 0.22.3.

**Reply by semiligneous** [reply]
@thouseef  - the link to "Startup Options" no longer exists...

I added `MEMOS_PUBLIC=false` to a .env file and reference it in docker-compose.yml as:

```
services:
  memos:
    image: neosmemo/memos:stable
    container_name: memos
    volumes:
      - ~/.memos/:/var/opt/memos
    env_file:
      - .env
```

Restarting the container still has the "sign up" option though.

Any ideas how this can be disabled?

Without the ability to disable sign up, there is literally no difference between "Public" and "Workspace" posts - since anyone can just sign up and see the posts anyway, so I am pretty confused how this is supposed to work!

thanks

**Reply by semiligneous** [reply]
@slyang40 - does this still work for you on latest version? Can you share your configuration (compose / .env etc)?

**Comment by semiligneous** [comment]
ahh - I found the option in the end.

in the UI (as and admin user) go Settings > System > Disallow user registration, and toggle that on.

I was looking everywhere for words "disable" or related to "sign up" - the wording being different made me miss it. Perhaps that wording could be re-considered in future?
